cli/mender-artifact/copy_test.go

```diff
@@ -166,59 +172,80 @@ func TestCopyRootfsImage(t *testing.T) {
 			verifyTestFunc: func(imgpath string) {
 				// as copy out of image is already verified to function, we
 				// will now use this functionality to confirm that the write
 				// was succesfull.
-				cmd := exec.Command(filepath.Join(dir, "mender-artifact"), "cat", imgpath+":/etc/mender/artifact_info")
+				cmd := exec.Command(filepath.Join(dir, "mender-artifact"), "cat", imgpath+":test.txt")
 				var out bytes.Buffer
 				cmd.Stdout = &out
 				err := cmd.Run()
-				assert.Nil(t, err, "got unexpected error: %v", err)
+				assert.Nil(t, err, "got unexpected error: %s", err)
 				assert.Equal(t, "artifact_name=foobar", out.String())
+				os.Remove("output.txt")
 			},
 		},
 		{
 			initfunc: func(imgpath string) {
 				assert.Nil(t, ioutil.WriteFile("input.txt", []byte("dummy-text"), 0755))
 			},
 			name: "copy file into a non-existing directory",
-			argv: []string{"mender-artifact", "cp", "input.txt", ":/non/existing/path"},
-			err:  "debugfs: error running command: \"cd /non/existing\\nclose\", err: \n/non/existing: File not found by ext2_lookup \n",
+			argv: []string{"mender-artifact", "cp", "input.txt", "<artifact|sdimg|fat-sdimg>:/non/existing/path"},
+			err:  "The directory: /non/existing does not exist in the image",
+			verifyTestFunc: func(string) {
+				os.Remove("input.txt")
+			},
 		},
 		{
 			name: "data on stdin",
-			argv: []string{"mender-artifact", "cp", ":/etc/mender/artifact_info", "-"},
+			argv: []string{"mender-artifact", "cp", "<artifact|sdimg|fat-sdimg>:/etc/mender/artifact_info", "-"},
 		},
 		{
-			name:        "write and read from the data partition",
-			argv:        []string{"mender-artifact", "cp", "output.txt", ":/data/test.txt"},
-			validimages: []int{1, 2}, // Not valid for an artifact.
+			name: "write and read from the data partition",
+			initfunc: func(imgpath string) {
+				assert.Nil(t, ioutil.WriteFile("artifact_info", []byte("artifact_name=foobar"), 0755))
+				_, err := os.Open("artifact_info")
+				require.Nil(t, err)
+			},
+			argv: []string{"mender-artifact", "cp", "artifact_info", "<sdimg|fat-sdimg>:/data/test.txt"},
 			verifyTestFunc: func(imgpath string) {
 				cmd := exec.Command(filepath.Join(dir, "mender-artifact"), "cat", imgpath+":/data/test.txt")
 				var out bytes.Buffer
 				cmd.Stdout = &out
 				err := cmd.Run()
 				require.Nil(t, err, "catting the copied file does not function")
 				assert.Equal(t, "artifact_name=foobar", out.String())
+				os.Remove("artifact_info")
 			},
 		},
 		{
-			name:        "error: artifact does not contain a data partition",
-			validimages: []int{0},
-			argv:        []string{"mender-artifact", "cp", "output.txt", ":/data/test.txt"},
-			err:         "newArtifactExtFile: A mender artifact does not contain a data partition, only a rootfs",
+			name: "error: artifact does not contain a data partition",
+			initfunc: func(string) {
+				assert.Nil(t, ioutil.WriteFile("output.txt", nil, 0755))
```

Comment:
This is very good: not to reuse an "output" from a previous test case

I suggest we call it "input.txt" ;)

Reply:
Here and in the following test cases

Reply:
Or foo.txt or something else. My point is not to be output as that has another meaning in many test cases